spec/models/partner_spec.rb

```diff
@@ -151,11 +151,57 @@
     end
   end
 
-  describe '#invite_new_partner' do
+  describe 'changing emails' do
     let(:partner) { create(:partner) }
 
-    it "should call the PartnerUser.invite! when the partner is changed" do
+    it "should not call the PartnerUser.invite! when the partner has status uninvited and the email is changed" do
       allow(PartnerUser).to receive(:invite!)
+      partner.status = :uninvited
+      partner.email = "randomtest@email.com"
+      partner.save!
+      expect(PartnerUser).not_to have_received(:invite!).with(
+        {email: "randomtest@email.com", partner: partner.profile}
+      )
+    end
+    it "should not call the PartnerUser.invite! when the partner has status deactivated and the email is changed" do
```

Comment:
A minor thing on readability, could you add a space between the `it` blocks? It makes it a little easier to read.